bigbite/themer
# [bbt-59] Refactor rest api

## Description

Completes [#59 ](https://github.com/bigbite/themer/issues/59) - This PR refactors the REST API to use the reccomended `wp_get_global_*` functions instead of the `WP_THEME_JSON_Resolver` methods to retrieve `theme.json` data.

It also changes the way themer checks for a compatible theme. Previously we checked if a `theme.json` existed in the theme but seeing as the presence (or lack) of this doesn't necessarily indicate a block theme, it wasn't reliable. 
This has now been changed to use `wp_is_block_theme()` instead, and will disable the plugin with a warning when not using a block theme.



## Change Log



- Uses `wp_get_global_*` functions to get data
- Disables plugin when block theme is not active

## Steps to test



- Build and `composer update`
- With a block theme active, make sure themer can be activated and deactivated without error
- Test the plugin still functions correctly. Everything should work as before.
- Switch to a classic theme while themer is active. Make sure the plugin disables itself with a notice.
- Try and activate themer with a classic theme. The same notice should appear and themer should not activate

Note: There is an edge case. When a classic theme is selected when themer is active, themer will be disabled but the 'styles editor' menu item will remain until the page is refreshed. I would welcome a solution to this but this is fine for now.

## Screenshots/Videos

http://bigbite.im/v/dY1RdF



## Checklist:
